### PR TITLE
feat: move back swc_ecma_dep_graph into deno_ast

### PR DESCRIPTION
**Pull request by dsherret**
Moves swc_ecma_dep_graph back into denoland code. I'll open a separate PR with improvements to this.

See https://github.com/swc-project/swc/discussions/8289

### CODE REVIEW

File: src/dep_graph.rs [hunk 1]
```diff
@@ -0,0 +1,849 @@
...
+  ImportType,
+  ImportEquals,
+  Export,
+  ExportType,
+  ExportEquals,
+  Require,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ImportAssertion {
```
Comment (dsherret) [inline src/dep_graph.rs:41]: I'll fix the naming of this in a separate PR.

**Review by lucacasonato — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ bundler = ["swc_bundler", "swc_ecma_transforms_optimization", "swc_graph_analyze
 cjs = ["utils", "visit"]
 codegen = ["swc_ecma_codegen", "swc_ecma_codegen_macros", "swc_macros_common"]
 compat = ["transforms", "swc_ecma_transforms_compat", "swc_trace_macro", "swc_config", "swc_config_macro"]
-dep_graph = ["swc_ecma_dep_graph"]
+dep_graph = ["visit"]
```
Comment (dsherret) [inline Cargo.toml:20]: Feel like "dep_graph" isn't a great name because it's not really a graph.

Comment (lucacasonato) [reply Cargo.toml:20]: dep_collect?

Comment (dsherret) [reply Cargo.toml:20]: I'll rename it in a future PR. Maybe dep_analysis

**Pull request merged**